python/mxnet/gluon/block.py

```diff
@@ -180,6 +180,22 @@ def __repr__(self):
 
     def __setattr__(self, name, value):
         """Registers parameters."""
+        def _find_block_in_nested(data):
+            # Find whether a nested structure contains Blocks
```

Comment:
This logic looks complicated and doesn't catch the case such as
```python
def __init__(self):
    self.blocks = []
    with self.name_scope():
        for i in range(3):
            self.blocks.append(...)
```
Instead of detecting whether a container has child blocks, you can detect for setattr with containers and give warning that anything in those containers won't be registered.

Reply:
There will be too many warnings if I do this. I decide to check the attributes inside collect_params